raft_batch_test.go

```diff
@@ -0,0 +1,26 @@
+// +build batchtest
+
+package raft
+
+func init() {
+	userSnapshotErrorsOnNoData = false
+}
+
+// NOTE: This is exposed for middleware testing purposes and is not a stable API
```

Comment:
We have a pattern in other modules to create a `testing.go` file that exposes Public API for test helpers for downstream dependencies. For example helpers to generate stub data or manipulate config or expose mocked out components.

I suggest we do the same here rather than have a test-time only dep documented this way.